pages/apis/graphql/graphql_resource_limits.md

```diff
@@ -0,0 +1,174 @@
+
+### Requested complexity
+The requested complexity is based on the number of fields and objects requested. Usually, requesting a deeply nested query or excluding pagination details from connections results in high requested complexity.
+
+A simple query like the following would incur more than 500 requested complexity points as the query asks for 503 possible resources.
+
+```graphql
+query RecentPipelineSlugs {
+  organization(slug: "organization-slug") {  # 1 point
+    pipelines(first: 500) {                  # 1 point
+      edges {                                # 1 point
+        node {                               # 500 points
+          slug                               # 0 point
+        }
+      }
+    }
+  }
+}
+```
+
+### Actual complexity
+The actual complexity is based on the results returned after the query execution, since the connection fields can return fewer nodes than requested. Lowering requested complexity usually lowers the actual complexity of queries.
+
+Taking the same query as above, if the organization has only 10 pipelines, the actual complexity will be around 13.
+
+```graphql
+query RecentPipelineSlugs {
+  organization(slug: "organization-slug") {  # 1 point
+    pipelines(first: 500) {                  # 1 point
+      edges {                                # 1 point
+        node {                               # 10 points
+          slug                               # 0 point
+        }
+      }
+    }
+  }
+}
+```
+
+<!-- How to show example -->
+
+
+
+## Rate limits
+Buildkite has implemented two distinct limits to our GraphQL endpoints. These limits play a critical role in ensuring that our platform operates smoothly and efficiently, while also minimising the risk of unnecessary downtime or system failures.
+
+By enforcing these limits, we can effectively manage and allocate the necessary resources for our GraphQL endpoints.
+
+### Single query limit
+
+Buildkite's API has a requested complexity limit of 50,000 for each individual query. This limit is enforced prior to query execution. The intention of this limit is to prevent users from requesting an excessive number of resources in a single query.
+
+As a best practice, we recommend breaking up queries into smaller, more manageable chunks and utilizing pagination to navigate through the resulting list, rather than relying on a single large query.
+
+If the query exceeds the limit, the response will return HTTP 200 status code with the following error.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Query has complexity of 251503, which exceeds max complexity of 50000"
+    }
+  ]
+}
+```
+
+
+### Time based rate limit
+
+To ensure optimal performance, an organization can use up to 20,000 actual complexity points within a 5 minute period. By allowing a set number of actual complexity points, we provide organizations flexibility to run queries of different sizes within a 5 minute window.
```

Comment:
I know we've talked about it before, but with a requested complexity limit of 50K and an Actual of 20K over 5 minutes, the first query can use 50K points and then subsequent queries will fail, is that correct?

Reply:
Yup!